src/devices/cpu/mipsx/mipsxdasm.cpp

```diff
@@ -0,0 +1,588 @@
+				}
+				else
+				{
+					util::stream_format(stream, "illegal asr form");
+				}
+			}
+			else
+			{
+				util::stream_format(stream, "unknown TY1 subcase 1 (%02x, %02x, %02x %04x)", get_src1(opcode), get_src2_dest(opcode), get_compute_dest(opcode), get_compute_compfunc(opcode));
+			}
+			break;
+		}
+
+		case 4:
+		{
+			u16 comp = get_compute_compfunc(opcode);
+
+			if (comp == 0x00b)
+			{
+				util::stream_format(stream, "blc %s, %s,%s", get_regname(get_src1(opcode)), get_regname(get_src2_dest(opcode)), get_regname(get_compute_dest(opcode)));
+			}
+			else if (comp == 0x00f)
+			{
+				if (get_src2_dest(opcode) == 0x00)
+				{
+					util::stream_format(stream, "not %s, %s", get_regname(get_src1(opcode)), get_regname(get_compute_dest(opcode)));
+				}
+				else
+				{
+					util::stream_format(stream, "illegal not form");
+				}
+			}
+			else if (comp == 0x019)
+			{
+				u8 src2 = get_src2_dest(opcode);
+				if (src2 == 0)
+				{
+					u8 src1 = get_src1(opcode);
+					u8 dest = get_compute_dest(opcode);
+
+					if ((src1 == 0) && (dest == 0))
+					{
+						util::stream_format(stream, "nop");
+					}
+					else
+					{
+						util::stream_format(stream, "mov %s, %s", get_regname(get_src1(opcode)), get_regname(get_compute_dest(opcode)));
+					}
+				}
+				else
+				{
+					util::stream_format(stream, "add %s, %s, %s", get_regname(get_src1(opcode)), get_regname(get_src2_dest(opcode)), get_regname(get_compute_dest(opcode)));
+				}
+			}
+			else if (comp == 0x01b)
+			{
+				util::stream_format(stream, "xor %s, %s, %s", get_regname(get_src1(opcode)), get_regname(get_src2_dest(opcode)), get_regname(get_compute_dest(opcode)));
+			}
+			else if (comp == 0x023)
+			{
+				util::stream_format(stream, "and %s, %s, %s", get_regname(get_src1(opcode)), get_regname(get_src2_dest(opcode)), get_regname(get_compute_dest(opcode)));
+			}
+			else if (comp == 0x026)
+			{
+				util::stream_format(stream, "subnc %s, %s, %s", get_regname(get_src1(opcode)), get_regname(get_src2_dest(opcode)), get_regname(get_compute_dest(opcode)));
+			}
+			else if (comp == 0x03b)
+			{
+				util::stream_format(stream, "or %s, %s, %s", get_regname(get_src1(opcode)), get_regname(get_src2_dest(opcode)), get_regname(get_compute_dest(opcode)));
+			}
+			else if (comp == 0x066)
+			{
+				util::stream_format(stream, "sub %s, %s, %s", get_regname(get_src1(opcode)), get_regname(get_src2_dest(opcode)), get_regname(get_compute_dest(opcode)));
+			}
+			else
+			{
+				util::stream_format(stream, "unknown compute (%02x, %02x, %02x %04x)", get_src1(opcode), get_src2_dest(opcode), get_compute_dest(opcode), get_compute_compfunc(opcode));
+			}
+			break;
+		}
+
+		default:
+		{
+			util::stream_format(stream, "Unhandled TY1 (%08x)", opcode);
+			break;
+		}
+		}
+		break;
+
+	}
+
+	case 2:
+	{
+		u8 op = get_op(opcode);
+		int imm17 = get_offset(opcode);
+		imm17 = util::sext(imm17, 17);
```

Comment:
Unsure if this was intentional or an oversight - since `get_offset()` starts from the LSB, you can just use `util::sext()` directly like with the displacement and remove `get_offset()` entirely.

Reply:
true, it's more just a byproduct of how the code was developed (first putting all the different fields behind get_ functions) so it's ended up a little more verbose than maybe neccessary.